### DISCUSSION TITLE
Folder comparison doesn't work

### DISCUSSION THREAD

**Discussion by gerd289** [WinMerge/winmerge#2134] [Q&A]
![example](https://github.com/WinMerge/winmerge/assets/120167632/4961e6ea-caef-4492-b743-eb185ab33908)
Hi,
I having a problem since some time, which makes me crazy.
After updating a very old WinMerge version to a recent one, the (folder) comparison doesn't work anymore.
For all files is shown "File skipped" and I need to double click each file to get it compared, which is a pain.
I probably do something but cannot find the issue myself. Does anybody have a hint how to get back to normal behavior?
Thanks in advance,
Gerd

**Comment by sdottaka** [comment]
In the attached image, the filter is `sam*.*`.
Please specify `*.*` in the "Select Files or Folders" window as shown below.

![image](https://github.com/WinMerge/winmerge/assets/98126/c90e325d-c136-4f06-81d7-88e436c85bd9)

**Reply by gerd289** [reply]
Thanks, this was helpful.

**Reply by hanjo** [reply]
Hi,
I just had the same issue, but my filter was just "`*`", which I think should be the correct choice over "`*.*`", because there might exist files that do not have a file extension are a thus missing the dot ("`.`"). However using just "`*`" results in the same result as above, which shows all results as "File skipped". Not sure why this is, but from a logical point of view, I believe the default should be just "`*`".

**Reply by sdottaka** [reply]
On Windows/DOS, unlike Linux/Unix, `*.*` has traditionally been a pattern that matches all files.

For this reason, WinMerge uses `*.*` as the default.

Currently, in WinMerge, if you specify `*` as Filter, it matches only files with no extension.

However, running `Dir *` at the command prompt has the same effect as `Dir *.*`, so it might be better to have `*.*` behave like that when `*` is specified.

**Comment by filsedla** [comment]
How do I achieve the same result when the folder comparison is initiated from the Windows Explorer context menu?

**Reply by sdottaka** [reply]
If you select only one folder in Explorer and launch WinMerge from the Explorer context menu, the Select Files or Folders window shown above will be displayed.
In this window, specify the second folder, specify `*.*` in Folder: Filter, and press the OK button.
